app/src/main/java/edu/mills/heartoftaiwanese/activity/favorites/FavoritesViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package edu.mills.heartoftaiwanese.activity.favorites
+
+import android.content.Context
+import edu.mills.heartoftaiwanese.activity.BaseFragment
+import edu.mills.heartoftaiwanese.activity.BaseViewModel
+import edu.mills.heartoftaiwanese.data.DatabaseWord
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+class FavoritesViewModel : BaseViewModel(), FavoritesContract.FavoritesViewModel {
+    private lateinit var view: FavoritesContract.FavoritesView
+
+    override fun configure(view: BaseFragment, context: Context) {
+        super.configure(view, context)
+        this.view = view as FavoritesFragment
+        this.view.onConfigurationSuccess()
+    }
+
+    override fun favoriteWord(word: DatabaseWord) {
+        GlobalScope.launch {
+            repository.favorite(word.id, !word.favorite)
+            getUpdatedWordList()
+        }
+    }
+
+    /**
+     * Get the updated word list. Return true if there are more than 0 words, false if there are none.
+     */
+    override fun getUpdatedWordList(): Boolean {
+        var returnValue = false
```

Comment:
A better name would be `isListEmpty`. Invert it before returning.

I think even better would be eliminating the variable and declaring `favoritesList` outside `GlobalScope.launch`. Then you could return `favoritesList.isNotEmpty()` and avoid the unnecessary variable and mutation.